grouper/log_redact.py

```diff
@@ -0,0 +1,42 @@
+from dataclasses import dataclass
+import logging
+import re
+
+@dataclass
+class RedactionPattern:
+    # regular expression to match for redaction
+    regex: str
+    # sensitive string will be replaced by this "redacted_message" in the log output
+    redacted_message: str
+
+class RedactingFormatter(logging.Formatter):
+    """
+    RedactingFormatter masks sensitive regular expressions from log entries.
+    
+    Configure the sensitive log regular expressions in settings.log_redact_patterns.
```

Comment:
Aren't these hardcoded now?

Reply:
Yeah. I'll rewrite the comment. I had originally thought to allow this in settings files, but I figured this would be easier and we'd gain little benefit from configuring it in the configs.